source/lib/vagrant-openstack-provider/client/nova.rb

```diff
@@ -175,6 +196,63 @@ def attach_volume(env, server_id, volume_id, device = nil)
         end
       end
 
+      # List snapshot images associated with a particular server
+      #
+      # @param env [Hash] Vagrant action environment
+      # @param server_id [String] Server UUID
+      #
+      # @return [Array<VagrantPlugins::Openstack::Domain::Image>]
+      def list_snapshots(env, server_id)
+        get_all_images(env, params: { server: server_id })
+      end
+
+      # Create a named snapsot for a given VM
+      #
+      # @param env [Hash] Vagrant action environment
+      # @param server_id [String] Server UUID
+      # @param snapshot_name [String]
+      #
+      # @return [void]
+      def create_snapshot(env, server_id, snapshot_name)
+        instance_exists do
+          post(
+            env,
+            "#{@session.endpoints[:compute]}/servers/#{server_id}/action",
+            { createImage: {
+              name: snapshot_name,
+              metadata: { vagrant_snapshot: true }
```

Comment:
It fails on `vagrant snapshot save`. The API respond a 400 error because of the boolean `true`. Replacing it by the string `"true"` fix it.

```
==> default: Snapshotting the machine as 'test-snap-01'...
2016-11-15 20:00 | DEBUG | block in create_snapshot - start
2016-11-15 20:00 | DEBUG | request  => method  : POST
2016-11-15 20:00 | DEBUG | request  => url     : http://openstack:8774/v2/f977fe8845974e64a643c9079bd40c3f/servers/86a4d73f-81e9-4ea4-bdf5-e359b1b16855/action
2016-11-15 20:00 | DEBUG | request  => headers : {"X-Auth-Token"=>"13ef9c8931974c0184c0bf62cc5f014b", :accept=>:json, :content_type=>:json}
2016-11-15 20:00 | DEBUG | request  => body    : {"createImage":{"name":"test-snap-01","metadata":{"vagrant_snapshot":true}}}
2016-11-15 20:00 | DEBUG | response => code    : 400
2016-11-15 20:00 | DEBUG | response => headers : {:content_length=>"137", :content_type=>"application/json; charset=UTF-8", :x_compute_request_id=>"req-a1a74fe0-fafe-486b-bc60-76984af9a0c2", :date=>"Tue, 15 Nov 2016 19:00:44 GMT"}
2016-11-15 20:00 | DEBUG | response => body    : {"badRequest": {"message": "Invalid input for field/attribute vagrant_snapshot. Value: True. True is not of type 'string'", "code": 400}}
```

@Sharpie Does it work in your tests ? Maybe an implementation difference on nova across different OpenStack versions. I ran my tests on an Canonical distribution based on Liberty.